fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseRunner.java

```diff
@@ -44,27 +44,8 @@
 import java.util.function.BiFunction;
 import java.util.function.Function;
 
-/**
- * A context for running against an {@link FDBDatabase} with retrying of transient exceptions.
- *
- * Implements {@link #run} and {@link #runAsync} methods for executing functions in a new {@link FDBRecordContext} and returning their result.
- *
- * Implements {@link #close} in such a way that {@code runAsync} will not accidentally do more work afterwards. In particular, in the case
- * of timeouts and transaction retries, it is otherwise possible for a whole new transaction to start at some time in the future and change
- * the database even after the caller has given up, due to a timeout, for example.
- *
- *
- * <pre><code>
- * try (FDBDatabaseRunner runner = fdb.newRunner()) {
- *     CompleteableFuture&lt;Void&lt; future = runner.runAsync(context -&gt; ...);
- *     fdb.asyncToSync(timer, WAIT_XXX, future);
- * }
- * </code></pre>
- *
- * @see FDBDatabase
- */
 @API(API.Status.MAINTAINED)
-public class FDBDatabaseRunner implements AutoCloseable {
+public class FDBDatabaseRunner implements FDBDatabaseRunnerInterface {
```

Comment:
Yes. Do we want to do that?

Reply:
Yes. Does it make sense to leave these breaking changes as a separate PR and do them when we have version bumps? This PR is going to be heavily packed after all.

Reply:
It might make sense. This is kind of cheating, but in theory, we could mark the constructor on `FDBDatabaseRunner` as `INTERNAL` on 2.7 and then make this change on 2.8.